File: UVa1595/.cpp
#pragma GCC optimize(3, "Ofast", "inline")
#include <bits/stdc++.h>
using namespace std;
map<int, vector<int>> pts;
set<int> left_plus_right;
int main()
{
    int T;
    cin >> T;
    int n;
    int x, y;
    while (T--)
    {
        left_plus_right.clear();
        pts.clear();
        for (cin >> n; n; cin >> x >> y, pts[y].push_back(x), n--)
            ;
        for (auto &i : pts)
        {
            auto pt = i.second;
            sort(pt.begin(), pt.end());
            if (pt.size() % 2 == 0)
            {
                for (size_t i = 0; i < pt.size() / 2; i++)
                {
                    left_plus_right.insert(*(pt.begin() + i) + *(pt.end() - i - 1));
                }
            }
            else
            {
                for (size_t i = 0; i < pt.size() / 2 + 1; i++)
                {
                    left_plus_right.insert(*(pt.begin() + i) + *(pt.end() - i - 1));
                }
            }
        }
        if (left_plus_right.size() == 1)
        {
            printf("YES\n");
        }
        else
        {
            printf("NO\n");
        }
    }
    return 0;
}

File: UVa1595/.py
from collections import defaultdict
T = int(input())
left_plus_right = set()
pts = defaultdict(list)
for i in range(T):
    left_plus_right.clear()
    pts.clear()
    n = int(input())
    for j in range(n):
        x, y = input().split()
        pts[int(y)].append(int(x))
    for pt in pts.values():
        pt.sort()
        for k in range(len(pt)//2+(len(pt) % 2)):
            left_plus_right.add(pt[k]+pt[-(k+1)])
    if len(left_plus_right) == 1:
        print('YES')
    else:
        print('NO')
